src/slice.c

```diff
@@ -8,7 +8,8 @@ SEXP fns_vec_slice_fallback = NULL;
 // Defined below
 SEXP vec_as_index(SEXP i, SEXP x);
 static void slice_names(SEXP x, SEXP to, SEXP index);
-static SEXP vec_slice_impl(SEXP x, SEXP index, bool dispatch);
+static SEXP vec_slice_impl(SEXP x, SEXP index, SEXP to, bool dispatch);
```

Comment:
I think it might be worth including `vec_slice_bare()` here as well (even though it's not technically needed) and including a longer comment that describes the differences between the variants (and when you would use them)